### ISSUE TITLE
item.genres difference between v2 and v3

### ISSUE THREAD

**Issue by amazingr4b** [pkkid/python-plexapi#249]
I noticed another subtle issue in the more recent versions of the plex api, though like the previous I'm not sure what version it would have started. 

When getting the genres from movie objects the list returned is only a partial list for some items. For example, "300" in my library has tags - Drama, War, Action, Fantasy. When I look at the list of genres in the movie object it is only showing 2 (drama and war).

Running the same actions using 2.0.2 and I get the entire list as expected.

I noticed this partial list issue when making use of the addGenre action. I added a "Favorites" genre to an item, and when I then checked the genre list I did not see it listed. When checking the Plex Web I noticed that not only was my new favorites genre absent in the list returned by the api, but so too were other genres. I have not checked every item in my library, but of those that I have checked this seems to affect titles that have more than 2 genre tags. 

Any ideas on this one? I can post a screen capture video to help illustrate the issue if my writeup above was hard to follow.

**Comment by Hellowlol** [comment]
We get as much info as we using as few http calls as possible. If you want more then 3 list items you probable need to do object.reload() (you dont see that you havnt added favorite for the same reason.)

**Comment by pkkid** [comment]
This is a tradeoff we made for performance. When you look at the xml returned when listing a url like movies/all, it will include a bunch of movie properties which does contain a list of genres for the movie, but not all of them. Plex probably did this because the XML being returned would be insanely massive for people with larger libraries.

**Comment by amazingr4b** [comment]
Now that you mention the /all page, I noticed the same behavior is showing in roles as well. Are you saying you changed the page it looks at to now get that data from /all instead of the one for the item?

**Comment by pkkid** [comment]
We used to try being smart about calling reload() without the programmer needing to call it explicitly. However, the logic was quite confusing and didn't always work as intended much of the time. This slowed down many scripts as we called reload() in the background when it was not always needed. We made the decision to rip it out and deal with explaining to people whats going on and how to call it only when needed.
